lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -476,8 +476,16 @@ sub mark_job_linked ($self, $jobid, $referer_url) {
     return undef if !$job || ($referer->path_query =~ /^\/?$/);
     my $comments = $job->comments;
     return undef if $comments->search({text => {like => 'label:linked%'}}, {rows => 1})->first;
+    return undef unless _specific_issue_exists($referer);
```

Comment:
As this is a referer-related check which only needs the url, would be better to place it just after checking the `$referer_host` (i.e. line 475) to spare ourselves some potentially costly database lookups (finding the job and searching comments)?

Reply:
It should not be its own check anyway. We do already check the `recognized_referers` in line 473, so I don't know why this was put in a different location.

Reply:
@b10n1k were you actually aware of `recognized_referers` before creating this PR?